### DISCUSSION TITLE
Aurelia2 branch

### DISCUSSION THREAD

**Discussion by MaximBalaganskiy** [ghiscoding/aurelia-slickgrid#1036] [General]
Just continuing where we left on discourse.

Not sure I understand the folder structure issue. I basically copied the plugin source into a fresh au2 plugin skeleton. So, with minor differences, the files should be in familiar places.

The skeleton is a sensible start in my opinion since it bears the core team best practices.

As for the upstream branch, I can definitely create a PR for it.

**Comment by ghiscoding** [comment]
The main issue with the structure is that all the imports are different in every examples, it's not a major issue but I'd like to keep the same structure as much as possible even if it deviates from the best practice. 

There's 1 TODO that you left in the constructor where you commented out the last argument `externalServices`, I was using this for unit tests only. I was manually constructing the Aurelia-Slickgrid element and passing these services by myself to make it easy for unit testing and I know it's not part of the Dependency Injection but it was working fine in Au1. I was also doing something similar in my other lib Angular-Slickgrid with `@Inject('externalService') externalServices`, do we have something similar in Au2 where I could inject my own stuff?

Thanks a lot for the help, it finally looks like it's going to happen :)
